hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java

```diff
@@ -4222,7 +4222,14 @@ public String explainFailure() throws IOException {
 
       @Override
       public boolean evaluate() throws IOException {
-        return getHBaseAdmin().tableExists(tableName) && getHBaseAdmin().isTableEnabled(tableName);
+        boolean tableEnabled = false;
+        try {
+          tableEnabled = getHBaseAdmin().tableExists(tableName)
+              && getHBaseAdmin().isTableEnabled(tableName);
+        } catch (TableNotFoundException tnfe) {
+          // Ignore TNFE
+        }
+        return tableEnabled;
```

Comment:
This could be
```
      public boolean evaluate() throws IOException {
        try {
          return getHBaseAdmin().tableExists(tableName)
              && getHBaseAdmin().isTableEnabled(tableName);
        } catch (TableNotFoundException tnfe) {
          return false;
        }
      }
```

Reply:
Thank you for the feedback. Done in latest commit. Please review again.